### PR TITLE
Add sparkly

### PR DESCRIPTION
**Pull request by veelenga**
The resource itself: https://github.com/Tubular/sparkly

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -100,6 +100,7 @@ A curated list of awesome [Apache Spark](https://spark.apache.org/) packages and
 ### Utilities
 
 * [silex](https://github.com/willb/silex) - A bunch of tools varying from ML extensions to additional RDD methods.
+* [sparkly](https://github.com/Tubular/sparkly) - Helpers & syntax sugar for PySpark
```
Comment (zero323) [inline README.md]: - Shouldn't it be _syntactic sugar_ instead of  _syntax sugar_?
- Missing full stop.

Comment (veelenga) [inline README.md]: Good point. Fixed

**Comment by zero323** [comment]
PR alone LGTM. Regarding the project I would like to get some response to https://github.com/Tubular/sparkly/issues/29 first. @eliasah?

**Comment by eliasah** [comment]
I agree with @zero323 on this

**Comment by veelenga** [comment]
Actually response is there. Please be free to ask if you need enything else. Is "Spark 2.x support" a must-have requirement to be a part of the list ? I found nothing about that in [README.md](https://github.com/awesome-spark/awesome-spark/blob/master/README.md) and [contributing.md](https://github.com/awesome-spark/awesome-spark/blob/master/contributing.md) either.

Thank you for the review and your work on the list :)

**Comment by zero323** [comment]
@veelenga It is not about must-have, more about common sense. It has been almost half a year since 2.0 release and ability to follow Spark development cycle is a good sign that project, especially relatively small one, hasn't stalled. Sadly, this is a common issue in the Spark ecosystem, and in hindsight we should have anticipate that. 

So what I am trying to avoid is adding a project I am not familiar with and haven't used in practice just to put in on to-remove list (this one tends to grow faster than the _awesome_ part) and I don't mean to offend anyone here :)

**Comment by veelenga** [comment]
@zero323 ok, i understand. Thank you for explanation.

**Comment by drudim** [comment]
@zero323 so, what are the requirements to become a part of the list?

**Comment by zero323** [comment]
@drudim Following the [contribution guidelines](https://github.com/awesome-spark/awesome-spark/blob/master/contributing.md) and some informal consensus between team members (this means me and @eliasah). 

I cannot speak for Eli, but if I am not familiar with a project,  it is important for me to be able to take it for a test run. It is curated list after all. This means that:

- Code has to look good enough for me to trust it in my own environment ( ✔).
- I have to be interested in the functionality (✔).
- It cannot depend on legacy environment (like JDK 7, Scala 2.10 or Python 2.x) ( ✔) or Spark version I haven't used for a while (1.x) (✘).  

If these criteria are not satisfied I usually trust Eli's judgment and experience with a given project. I recognize these are highly  subjective criteria, but as someone told me lately, _open source project allows expressing opinions through forks_.

**Comment by eliasah** [comment]
Unfortunately, like mentioned by @zero323, adding dead project doesn't help anyone. Some needs to be tested and other deprecated. The whole idea of the list is to get something curated which actual people can use.

**Comment by zero323** [comment]
@drudim awesome aside, do you have any TODO list for 2.0?

**Comment by drudim** [comment]
@zero323 I am integrating Spark 2.x to our infrastructure right now and going to switch to `sparkly` from the next week. We don't want to distribute `sparkly` unless at least one of our internal services is ported to the release candidate. I will let you know as soon as PR for Spark 2.x is ready.

Thank you guys for taking care of the list!

**Comment by zero323** [comment]
@veelenga @drudim @skliarpawlo I had some time to play with the project and I think we can merge this PR. I you make me regret that I know where to find you 😈

**Comment by veelenga** [comment]
:D Thank you!

**Comment by drudim** [comment]
@zero323 support of spark 2.x is coming. I have successfully migrated one of our internal service to Spark 2.x this week using https://github.com/Tubular/sparkly/pull/35 I guess in a week or two, we will be ready to distribute it on PyPi.

**Pull request merged**